src/cmd_sym.ml

```diff
@@ -197,50 +197,58 @@ let cmd profiling debug unsafe optimize workers no_stop_at_failure no_values
   let result = List.fold_left (run_file ~unsafe ~optimize) pc files in
   let thread : Thread.t = Thread.create () in
   let results = Choice.run ~workers result thread in
-  let* failing =
-    list_fold_left
-      (fun failing (result, thread) ->
-        let pc = Thread.pc thread in
-        let symbols = thread.symbol_set in
-        let model = get_model ~symbols solver pc in
-        let* result, err =
-          match result with
-          | Symbolic_choice.Multicore.EVal (Ok ()) -> Ok (failing, false)
-          | EAssert assertion ->
-            Ok (`EAssert (assertion, model) :: failing, true)
-          | ETrap tr -> Ok (`ETrap (tr, model) :: failing, true)
-          | EVal (Error e) -> Error e
-        in
-        if not no_values then
-          let testcase =
-            List.sort compare (Encoding.Model.get_bindings model)
-            |> List.map snd
-          in
-          let+ () = write_testcase ~dir:workspace ~err testcase in
-          result
-        else Ok result )
-      [] (List.of_seq results)
-  in
   let print_bug = function
     | `ETrap (tr, model) ->
       Format.pp_std "Trap: %s@\n" (Trap.to_string tr);
       Format.pp_std "Model:@\n  @[<v>%a@]@\n"
         (Encoding.Model.pp ~no_values)
-        model
+        model;
+      Format.pp_flush Stdlib.Format.std_formatter ()
     | `EAssert (assertion, model) ->
       Format.pp_std "Assert failure: %a@\n" Expr.pp assertion;
       Format.pp_std "Model:@\n  @[<v>%a@]@\n"
         (Encoding.Model.pp ~no_values)
-        model
+        model;
+      Format.pp_flush Stdlib.Format.std_formatter ()
+  in
+  let rec print_and_count_failures count results =
+    match results () with
+    | Seq.Nil -> Ok count
+    | Seq.Cons ((result, thread), tl) ->
+      let pc = Thread.pc thread in
+      let symbols = thread.symbol_set in
+      let model = get_model ~symbols solver pc in
+      let* count, err =
+        match result with
+        | Symbolic_choice.Multicore.EAssert assertion ->
+          print_bug (`EAssert (assertion, model));
+          Ok (succ count, true)
+        | ETrap tr ->
+          print_bug (`ETrap (tr, model));
+          Ok (succ count, true)
+        | EVal (Ok ()) -> Ok (count, false)
```

Comment:
`count` is incremented if and only if `is_err == true` so this could be factored out, which would makes the intent clearer IMO.